docs/reference/security/external-principals-aws.md

```diff
@@ -120,11 +120,33 @@ for p in resp.results:
     # do something
 ```
 
+
 ## Get lakeFS API Token
 
 The login to lakeFS is done by calling the [login API][login-api] with the `GetCallerIdentity` request signed by the client.
 Currently, the login operation is supported out of the box in [lakeFS Hadoop FileSystem][lakefs-hadoopfs] version 0.2.4, see [Spark usage][lakefs-spark].
-Other clients (i.e HTTP, Python etc) can use the login endpoint to authenticate to lakeFS but, you will have to build the request input.
+and in [python](#login-with-python), for other use cases you can use the login endpoint to authenticate to lakeFS but, you will have to build the request input.
+
+## Login with python
+
+### prerequisites
+
+lakeFS requires additional python packages to be installed in order to generate a lakeFS client with the assumed role.
+To install the required packages, run the following command:
+
+```sh
+  pip install lakefs[aws-iam]
+```
+
+In order to generate a lakeFS client with the assumed role, initiate a boto3 session with the desired role and call the `get_caller_identity` method to get the caller identity:
+
+
+```python
+import lakefs
+import boto3    
+session = boto3.Session(profile_name='my-profile', region_name='us-west-2')
```

Comment:
let's not use the `region_name` because it sets the user for failure since we don't show them how to run the python. 

Only if they use the env `AWS_STS_REGIONAL_ENDPOINTS` when the run i.e `AWS_STS_REGIONAL_ENDPOINTS=regional python main.py`  the regional `us-west-2` will kick in. 

So either elaborate on that in the example or just remove the region when creating the session.

Reply:
removed